tests/unit/drivers/test_vector_index_driver.py

```diff
@@ -0,0 +1,180 @@
+from copy import deepcopy
+from typing import Iterator, Union, List, Tuple
+
+import numpy as np
+import pytest
+
+from jina import DocumentSet
+from jina.drivers.index import VectorIndexDriver
+from jina.executors.indexers import BaseVectorIndexer
+from jina.types.document import Document
+
+
+class MockGroundTruthVectorIndexer(BaseVectorIndexer):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.docs = {}
+
+    def add(self, keys: 'np.ndarray', vectors: 'np.ndarray', *args, **kwargs):
+        for key, value in zip(keys, vectors):
+            self.docs[key] = value
+
+    def update(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs):
```

Comment:
are also these used?

Reply:
yes in line 63

Reply:
where?

Reply:
@pytest.fixture(scope='function')
def simple_kv_indexer_driver_update():
    return SimpleVectorIndexDriver(method='update')


@pytest.fixture(scope='function')
def simple_kv_indexer_driver_delete():
    return SimpleVectorIndexDriver(method='delete')

Reply:
but this is `MockGroundTruthVectorIndexer` not `SimpleVectorIndexDriver`

Reply:
True, but the executor is attached to that driver and sets the `_exec` function accordingly.
`simple_vector_indexer_driver_add.attach(executor=mock_groundtruth_indexer, pea=None)`